### PR TITLE
plugin/kubernetes: update to client-go 12.0.0

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. Why is this pull request needed and what does it do?

Updates client-go to 12.0.0 (released with k8s 1.15).

### 2. Which issues (if any) are related?
none

### 3. Which documentation changes (if any) need to be made?
none

### 4. Does this introduce a backward incompatible change or deprecation?
no

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -59,13 +58,12 @@ require (
 	google.golang.org/grpc v1.22.0
 	gopkg.in/DataDog/dd-trace-go.v1 v1.16.0
 	gopkg.in/inf.v0 v0.9.1 // indirect
-	k8s.io/api v0.0.0-20190313235455-40a48860b5ab
-	k8s.io/apimachinery v0.0.0-20190313205120-d7deff9243b1
-	k8s.io/client-go v11.0.0+incompatible
+	k8s.io/api v0.0.0-20190620084959-7cf5895f2711
+	k8s.io/apimachinery v0.0.0-20190612205821-1799e75a0719
+	k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
```
Comment (rajansandeep) [inline go.mod:63]: Is this taking the dependency from `latest/master` instead of `12.0.0`?

**Comment by chrisohaver** [comment]
Not sure why go.mod isn't tagging client-go on the tag that I specify in go get (v12.0.0)... but the commit in the go.mod _is_ the same as the v12.0.0 tag... so all is good I guess.

**Review by yongtang — APPROVED** [review]
lgtm

File: go.mod (hunk 1 above)
Comment (chrisohaver) [reply go.mod:63]: No.  The latest commit in client-go is "f16ff99...", the tag for v12.0.0 is "78d2af7..."

Comment (rajansandeep) [reply go.mod:63]: Yeah, seems to be normal as described in https://github.com/kubernetes/client-go/issues/631

**Pull request merged**